File: medium/0442_find_all_duplicates_in_an_array/solution.py
from typing import List


class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        nums.sort()

        duplicates = []

        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                duplicates.append(nums[i])

        return duplicates

File: medium/0442_find_all_duplicates_in_an_array/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> findDuplicates(std::vector<int> &nums) {
    std::vector<int> duplicates;

    for (const int &num : nums) {
      int idx = abs(num) - 1;

      if (nums[idx] < 0)
        duplicates.push_back(idx + 1);
      else
        nums[idx] *= -1;
    }

    return duplicates;
  }
};
